File: TipCalculator/app/src/main/java/com/example/tipcalculator/MainActivity.java
package com.example.tipcalculator;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.example.tipcalculator.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding mainBinding;
    private View view;
    private static final String TAG = "StateChange";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mainBinding = ActivityMainBinding.inflate(getLayoutInflater());
        View view = mainBinding.getRoot();
        setContentView(view);

        Log.i(TAG, "onCreate");
    }

    public void calculateTip(View view) {
        this.view = view;
        String prompt = "The Tip values are as follows";
        String tenPercentText = "Ten percent = $";
        String fifteenPercentText = "Fifteen percent = $";
        String twentyPercentText = "Twenty percent = $";
        String emptyField = "You must enter an amount";
        if(!mainBinding.dollarText.getText().toString().equals("")){
            float dollarValue = Float.valueOf(mainBinding.dollarText.getText().toString());
            String tenPercentValue = String.format("%.2f", dollarValue * .10F);
            String fifteenPercentValue = String.format("%.2f", dollarValue * .15F);
            String twentyPercentValue = String.format("%.2f", dollarValue * .20F);

            mainBinding.textView.setText(String.format("%s\n\n%s%s\n%s%s\n%s%s", prompt, tenPercentText,
                    tenPercentValue, fifteenPercentText, fifteenPercentValue, twentyPercentText,
                    twentyPercentValue));
        }
        else {
            mainBinding.textView.setText(emptyField);
        }
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        Log.i(TAG, "onSaveInstanceState");

        CharSequence userText = mainBinding.dollarText.getText();
        CharSequence tipVal = mainBinding.textView.getText();
        outState.putCharSequence("savedText", userText);
        outState.putCharSequence("savedCalculation", tipVal);
    }

    @Override
    protected void onRestoreInstanceState(Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);
        Log.i(TAG, "onRestoreInstanceState");
        CharSequence userText =
                savedInstanceState.getCharSequence("savedText");
        CharSequence tipVal =
                savedInstanceState.getCharSequence("savedCalculation");

        mainBinding.dollarText.setText(userText);
        mainBinding.textView.setText(tipVal);
    }
}

File: TipCalculator_Kotlin/app/src/main/java/com/cps251/tipcalculator/MainActivity.kt
package com.cps251.tipcalculator

import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.cps251.tipcalculator.databinding.ActivityMainBinding


open class MainActivity : AppCompatActivity() {
    lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
    }

    fun calculateTip(view: View) {
        var prompt = "The tip values are as follows";
        var tenPercentText = "Ten percent = $";
        var fifteenPercentText = "Fifteen percent = $";
        var twentyPercentText = "Twenty percent = $";
        var emptyField = "You must enter an amount";
        if(binding.dollarText.text.toString() != ""){
            var dollarValue = binding.dollarText.text.toString()
            var tenPercentValue = String.format("%.2f", dollarValue.toFloat() * .10F)
            var fifteenPercentValue = String.format("%.2f", dollarValue.toFloat() * .15F)
            var twentyPercentValue = String.format("%.2f", dollarValue.toFloat() * .20F)

            binding.textView.text = "$prompt\n\n$tenPercentText$tenPercentValue\n$fifteenPercentText" +
                    "$fifteenPercentValue\n$twentyPercentText$twentyPercentValue"
        } else {
            binding.textView.text = emptyField
        }
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        Log.i("thing", "onSaveInstanceState")
        val userText: CharSequence = binding.dollarText.text
        val tipVal: CharSequence = binding.textView.text
        outState.putCharSequence("savedText", userText)
        outState.putCharSequence("savedCalculation", tipVal)
    }

    @Override
    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
        Log.i("thing", "onRestoreInstanceState")
        val userText = savedInstanceState.getCharSequence("savedText")
        val tipVal = savedInstanceState.getCharSequence("savedCalculation")

        binding.dollarText.setText(userText)
        binding.textView.text = tipVal
    }
}